### PR TITLE
features/hdf-18 

### PR DESCRIPTION
**Pull request by ernestpascual**
- new HaloHalo abi
- made APY value and price dynamic
- modified useHaloHalo to support new HaloHalo contract version

How to test:
Make sure you have the latest env file. (updated one pass)
Navigate to `/vesting` 
Make sure everything loads properly

For the calculation cc: @ctverceles . You can check the useHaloHalo hook for more details
```
const currentBlockNumber = await providers?.getDefaultProvider().getBlockNumber()
const genesisTimestamp = Number(await halohaloContract?.genesisTimestamp())
const currentTimestamp = await (await providers?.getDefaultProvider().getBlock(currentBlockNumber)).timestamp
// one year in seconds / 31536000
const timeElapsed = currentTimestamp - genesisTimestamp
// 1 comes from the 1:1 ratio before adding rewards
const priceChange = +haloHaloPrice - 1
const APY = timeElapsed * priceChange
```

If ever you need to check the calculations:
```
HaloHalo Total Supply:  173.642857142857142855
Current HALOHALO Price:  3.230769230769230769
Total HALO in the contract:  561.000000000000000003
Genesis Timestamp:  1619549932

```

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export const HALO_REWARDS_ADDRESS: ChainAddressMap = {
 
 // HALO Chest Addressess
```
Comment (bitcoinbrisbane) [inline src/constants/index.ts]:
```suggestion
// HALOHALO Addressess
```

File: src/halo-hooks/useHaloHalo.ts [hunk 2]
```diff
@@ -16,71 +17,99 @@ const useHaloHalo = () => {
...
 
   const leave = useCallback(
     // todo: this should be updated with BigNumber as opposed to string
     async (amount: BalanceProps | undefined) => {
       if (amount?.value) {
         try {
-          const tx = await chestContract?.leave(amount?.value)
-          //const tx = await chestContract?.leave(ethers.utils.parseUnits(amount)) // where amount is string
+          const tx = await halohaloContract?.leave(amount?.value)
+          //const tx = await halohaloContract?.leave(ethers.utils.parseUnits(amount)) // where amount is string
```
Comment (bitcoinbrisbane) [inline src/halo-hooks/useHaloHalo.ts]: Can we remove?

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Few things to change, plus the linting issues.

File: src/halo-hooks/useHaloHalo.ts (hunk 2 above)
Comment (ernestpascual) [reply src/halo-hooks/useHaloHalo.ts]: thanks for spotting! will remove

**Comment by ernestpascual** [comment]
Updated: 
- number formatting
- better managment of APY calculation

**Comment by schystz** [comment]
Checked the code and tested this locally too - worked and looks good to me! :) Nice job @ernestpascual !

I have some feedback on how to improve UX but I'll add it in this Jira task instead https://halodao.atlassian.net/browse/HDF-164 since it has nothing to do with the changes in this PR.

Also, I'll leave the verification of APY & HALOHALO:HALO ratio calculation to @ctverceles

**Comment by schystz** [comment]
Actually, I just noticed that integration tests is failing for this PR. Can you take care of it too @ernestpascual ? Please refer to `INTEGRATION_TESTS.md` for howto, but you can also ping me for immediate help.

**Comment by ernestpascual** [comment]
Updated cypress test. All things are passing on my end @schystz

File: cypress/integration/vesting.test.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ describe('Vesting', () => {
 
   it('displays vesting APY', () => {
     // Initially APY blank
-    cy.contains('% APY')
+    cy.contains('APY')
 
     // Then should reload APY
     // cy.should('not.contain', '% APY')
```
Comment (schystz) [inline cypress/integration/vesting.test.ts]: @ernestpascual what I envision for this test actually is that "% APY" is displayed initially until APY is calculated, in that case "% APY" is replaced with "0.31% APY" (or to write this simply, just uncomment line #11 above)

Comment (schystz) [reply cypress/integration/vesting.test.ts]: Can we also add a similar test for line #14 below? So, initially check that ratio is undefined or 0, then should update to non-undefined/zero

**Review by schystz — COMMENTED** [review]
@ernestpascual a couple of suggestions for integration testing of Vesting page

**Comment by ernestpascual** [comment]
added additional tests and everything is passing. @schystz

**Review by schystz — APPROVED** [review]
Thanks @ernestpascual !

File: src/constants/index.ts [hunk 4]
```diff
@@ -82,9 +81,9 @@ export const HALO_REWARDS_ADDRESS: ChainAddressMap = {
   [ChainId.KOVAN]: process.env.REACT_APP_HALO_REWARDS_ADDRESS_KOVAN || ''
 }
 
-// HALO Chest Addressess
+// HALOHALO Addressess
 export const HALOHALO_ADDRESS: ChainAddressMap = {
-  [ChainId.KOVAN]: process.env.REACT_APP_HALOCHEST_ADDRESS_KOVAN || ''
+  [ChainId.KOVAN]: process.env.REACT_APP_HALOHALO_ADDRESS_KOVAN || ''
```
Comment (tracyarciaga) [inline src/constants/index.ts:86]: @ernestpascual Super minor request, can u also add the REACT_APP_HALOHALO_ADDRESS_KOVAN in the env.example file.

**Review by tracyarciaga — CHANGES_REQUESTED** [review]
Approving this with just minor request.

**Comment by ernestpascual** [comment]
done @tracyarciaga !

**Review by chris247474 — APPROVED** [review]
So (using the `halodao-interface.env` file here https://halodao.1password.com/vaults/gllrg2so74fsgd43baj5q5voui/allitems/nb3dl4j5frbhlleg2vn4wuw6ay) I ran the UI locally while pointed to kovan and checked the total HALOHALO supply here https://kovan.etherscan.io/token/0x4507557aFd5109Fd09cD18B8BFFa625b2c37C7b5#balances. Based on that, HALOHALO:HALO price = 723/(166.67 + 46.43 + 10.38 + 0.3) = 3.2308517... which roughly equals the calculation on the preview link 
image

The APY i guess is a bit harder to double check atm. i'll need to run the contracts and UI locally for a day to confirm, but the code does make sense and the APY value on the Vesting page seems to make sense when i do the math. @bitcoinbrisbane @tracyarciaga given this, do we ;

- leave it under Peer Review and not yet merge 
- or move to Testing and merge this?

**Comment by schystz** [comment]
@ctverceles @tracyarciaga I vote to merge this PR, so we can proceed with re-initing the repo, and me submitting my PR for HDF-152 which can be done today.

**Comment by tracyarciaga** [comment]
> * leave it under Peer Review and not yet merge

@ctverceles @ernestpascual I just read @schystz comment.  With that, let's just merge it and move to it testing. So James can fast-track his PR too.

**Pull request merged**